api/api.go

```diff
@@ -21,6 +17,36 @@ import (
 	"github.com/rubyist/tracerx"
 )
 
+// BatchOrLegacy calls the Batch API and falls back on the Legacy API
+// This is for simplicity, legacy route is not most optimal (serial)
+// TODO LEGACY API: remove when legacy API removed
+func BatchOrLegacy(objects []*ObjectResource, operation string) ([]*ObjectResource, error) {
```

Comment:
I actually want us to stop doing this in 1.3 & remove the legacy API. The persistent `lfs.batch=false` setting is a regular source of support calls when something has gone wrong either on the server, or some step in between and then persistently the LFS client refuses to use batch any more. If we keep `lfs.batch` in 1.3 then I'll push for adding a big warning banner in every fetch/push/pull that tells people `lfs.batch=false` and suggests removing it. What we do with batch/legacy is outside the scope of this PR though.